pkg/ddevapp/codeigniter.go

```diff
@@ -0,0 +1,185 @@
+		return "# Database omitted by DDEV configuration"
+	}
+
+	// Simple mapping: default to MySQL/MariaDB; switch to Postgres when requested.
+	driver := "MySQLi"
+	port := 3306
+	if app != nil && app.Database.Type == nodeps.Postgres {
+		driver = "Postgre"
+		port = 5432
+	}
+
+	// Standard DDEV credentials/hostnames
+	return fmt.Sprintf(`
+# Database Configuration
+database.default.hostname = db
+database.default.database = db
+database.default.username = db
+database.default.password = db
+database.default.DBDriver = %s
+database.default.DBPrefix =
+database.default.port = %d
+	`, driver, port)
+}
+
+// isDBOmitted returns true if "db" is in omit_containers.
+func isDBOmitted(app *DdevApp) bool {
+	if app == nil {
+		return false
+	}
+	for _, s := range app.OmitContainers {
+		if s == "db" {
+			return true
+		}
+	}
+	return false
+}
+
+// getCodeIgniterUploadDirs returns upload directories for CodeIgniter 4
+func getCodeIgniterUploadDirs(_ *DdevApp) []string {
+	return []string{"writable/uploads"}
+}
+
+// getCodeIgniterHooks returns example hook comments for CodeIgniter
+func getCodeIgniterHooks() []byte {
+	return []byte(`#  post-start:
+#    - exec: php spark migrate
+#    - exec: php spark db:seed`)
+}
+
+// setCodeIgniterSiteSettingsPaths sets the paths to settings files
+func setCodeIgniterSiteSettingsPaths(app *DdevApp) {
+	app.SiteSettingsPath = filepath.Join(app.AppRoot, ".env")
+	app.SiteDdevSettingsFile = ""
+}
+
+// isCodeIgniterApp detects if this is a CodeIgniter 4 application
+func isCodeIgniterApp(app *DdevApp) bool {
+	// Check for CodeIgniter 4 specific files
+	sparkPath := filepath.Join(app.AppRoot, "spark")
+	appConfigPath := filepath.Join(app.AppRoot, "app", "Config", "App.php")
+	publicIndexPath := filepath.Join(app.AppRoot, "public", "index.php")
+
+	return fileutil.FileExists(sparkPath) &&
+		fileutil.FileExists(appConfigPath) &&
+		fileutil.FileExists(publicIndexPath)
+}
+
+// codeIgniterPostStartAction runs after container start
+func codeIgniterPostStartAction(app *DdevApp) error {
+	// Ensure writable directory has correct permissions
+	writableDir := "writable"
+
+	_, _, err := app.Exec(&ExecOpts{
+		Service: "web",
+		Cmd:     fmt.Sprintf("chmod -R 775 %s", writableDir),
+	})
+
+	if err != nil {
+		util.Warning("Unable to set permissions on %s directory: %v", writableDir, err)
+	}
+
+	return nil
+}
+
+// codeIgniterImportFilesAction handles file imports
+func codeIgniterImportFilesAction(app *DdevApp, uploadDir, importPath, extPath string) error {
+	destPath := app.calculateHostUploadDirFullPath(uploadDir)
+
+	// Ensure destination directory exists
+	err := os.MkdirAll(destPath, 0755)
+	if err != nil {
+		return err
+	}
+
+	// Copy files from import path to destination
+	if isTar(importPath) {
+		err = archive.Untar(importPath, destPath, extPath)
+	} else {
+		err = fileutil.CopyDir(importPath, destPath)
+	}
+
+	return err
+}
+
+func getCodeIgniterComposerCreateAllowedPaths(app *DdevApp) ([]string, error) {
```

Comment:
Please add a comment on these explaining why they're allowed. Shouldn't most of these be replaced on composer create-project?